nameko/testing/waiting.py

```diff
@@ -0,0 +1,63 @@
+""" https://github.com/mattbennett/call-waiting
+"""
+import sys
+from contextlib import contextmanager
+from threading import Semaphore
+
+import six
+from mock import patch
+
+
+class WaitResult(object):
+    res = None
+    exc_info = None
+
+    @property
+    def has_result(self):
+        return self.res is not None or self.exc_info is not None
+
+    def send(self, res, exc_info):
+        if not self.has_result:
+            self.res = res
+            self.exc_info = exc_info
+
+    def get(self):
+        if self.exc_info is not None:
+            six.reraise(*self.exc_info)
+        return self.res
+
+
+@contextmanager
+def wait_for_call(obj, target, callback=None):
```

Comment:
do use have use-cases in mind for this helper on its own, beyond the entrypoint waiter? (having it standalone is fine in itself, but slightly concerned about growing our public api. in general we've not been very good about clarifying which helpers are internal and which are considered public)